### ISSUE TITLE
415 Error on POST with Request Object

### ISSUE THREAD

**Issue by ssindelar** [awslabs/aws-serverless-java-container#282]
* Framework version: 1.3.2
* Implementations: Jersey
* OS: Windows 10

## Scenario

I want to implement a POST requests that expects a JSON object in the Request Body like in the Pet Store example. To verify I checked out the Pet-Store jersey sample, but it doesn't work either.

## Expected behavior

The Request body should be mapped to the Method Parameter of the POST method.

## Actual behavior

When an Custom Object is expected as parameter the application returns an 415 Unsupported Media Type Status Code. Expecting a String works. It also works when I run an Unit test that does the request, but not when it is locally deployed using SAM.

## Steps to reproduce

1. Check out and run the Jersey Pet Store example
2. Do a POST Request to add a Pet

## Full log output

```
* Preparing request to http://localhost:3000/pets/
* Using libcurl/7.57.0-DEV OpenSSL/1.0.2o zlib/1.2.11 libssh2/1.7.0_DEV
* Current time is 2019-09-27T10:11:37.598Z
* Disable timeout
* Enable automatic URL encoding
* Disable SSL validation
* Enable cookie sending with jar of 0 cookies
* Found bundle for host localhost: 0x1a39a0b9680 [can pipeline]
* Re-using existing connection! (#4) with host localhost
* Connected to localhost (127.0.0.1) port 3000 (#4)

> POST /pets/ HTTP/1.1
> Host: localhost:3000
> User-Agent: insomnia/6.6.2
> Content-Type: application/json
> Accept: */*
> Content-Length: 135

| {
|     "id": "bfbfbb42-2198-4e5d-99b9-03533a09c123",
|     "breed": "Dalmatian",
|     "name": "Bella",
|     "dateOfBirth": 1209031368188
|   }

* upload completely sent off: 135 out of 135 bytes

< HTTP/1.1 415 Unsupported Media Type
< Content-Type: application/json
< Date: Fri, 27 Sep 2019 10:11:40 GMT
< Content-Length: 0


* Connection #4 to host localhost left intact
```

**Comment by sapessi** [comment]
Is this running through the SAM CLI @ssindelar?

Jersey returns 415 (as per the [Jax Rs specifications](https://docs.oracle.com/cd/E19798-01/821-1841/gipzh/index.html#6nmq2cp2i)) when it cannot find a resource method that matches the the request content type - ie has a `@Consumes` annotation that matches the request's content type. The sample clearly [has the correct annotation](https://github.com/awslabs/aws-serverless-java-container/blob/master/samples/jersey/pet-store/src/main/java/com/amazonaws/serverless/sample/jersey/PetsResource.java#L28) and your request is sending the correct `Content-Type` header. This leads me to believe that for some reason Jersey does not have access to the request headers. Up until not too long ago, there was a [well documented issue with SAM CLI](https://github.com/awslabs/aws-sam-cli/issues/830) not supporting API Gateway's new `multiValueHeaders` property. The fix for this was released in version [`0.17.0` of SAM CLI](https://github.com/awslabs/aws-sam-cli/releases/tag/v0.17.0). 

The whole long detour is to say: If you are using SAM CLI, could you make sure that you are running at least version `0.17.0`?

**Comment by ssindelar** [comment]
I'm sorry. I followed the README.md and installed SAM through npm. But instead of a new version it installed 0.2.11.

With the current SAM version it works fine.
